document/src/main/java/com/ritense/document/service/impl/JsonSchemaDocumentSearchService.java

```diff
@@ -165,40 +154,45 @@ private Page<JsonSchemaDocument> search(QueryWhereBuilder queryWhereBuilder, Pag
                 .setMaxResults(pageable.getPageSize());
         }
 
+        return new PageImpl<>(typedQuery.getResultList(), pageable, count(queryWhereBuilder));
+    }
+
+    private Long count(
+        QueryWhereBuilder queryWhereBuilder
+    ) {
+        final CriteriaBuilder cb = entityManager.getCriteriaBuilder();
         final CriteriaQuery<Long> countQuery = cb.createQuery(Long.class);
         Root<JsonSchemaDocument> countRoot = countQuery.from(JsonSchemaDocument.class);
         countQuery.select(cb.count(countRoot));
-        queryWhereBuilder.apply(cb, countQuery, selectRoot);
+        queryWhereBuilder.apply(cb, countQuery, countRoot);
 
+        // TODO: Should be turned into a subquery, and then do a count over the results from the subquery.
         List<Long> countResultList = entityManager.createQuery(countQuery).getResultList();
 
         Long count = 0L;
 
         if (!countResultList.isEmpty()) {
-            count = countResultList.get(0);
+            count = (long) countResultList.size();
```

Comment:
Wont this always result in `1`? There's usually only one result in a count query.

Reply:
Unless there's a group by, unfortunately.